tests/upload_test.go

```diff
@@ -281,3 +282,146 @@ var _ = Describe("Block PV upload Test", func() {
 		Entry("[posneg:negative][test_id:1369]fail given an invalid token (block)", false, http.StatusUnauthorized),
 	)
 })
+
+var _ = Describe("Namespace with quota", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	var (
+		orgConfig      *v1.ResourceRequirements
+		pvc            *v1.PersistentVolumeClaim
+		uploadProxyURL string
+		portForwardCmd *exec.Cmd
+		err            error
+	)
+
+	BeforeEach(func() {
+		By("Capturing original CDIConfig state")
+		config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		orgConfig = config.Spec.PodResourceRequirements
+		if pvc != nil {
+			By("Making sure no pvc exists")
+			Eventually(func() bool {
+				// Make sure the pvc doesn't still exist. The after each should have called delete.
+				_, err := f.FindPVC(pvc.Name)
+				return err != nil
+			}, timeout, pollingInterval).Should(BeTrue())
+		}
+	})
+
+	AfterEach(func() {
+		By("Restoring CDIConfig to original state")
+		config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		config.Spec.PodResourceRequirements = orgConfig
+		_, err = f.CdiClient.CdiV1alpha1().CDIConfigs().Update(config)
+		Expect(err).ToNot(HaveOccurred())
+		By("Stop port forwarding")
+		if portForwardCmd != nil {
+			err = portForwardCmd.Process.Kill()
+			Expect(err).ToNot(HaveOccurred())
+			portForwardCmd.Wait()
+			portForwardCmd = nil
+		}
+
+		By("Delete upload PVC")
+		err = f.DeletePVC(pvc)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Wait for upload pod to be deleted")
+		deleted, err := utils.WaitPodDeleted(f.K8sClient, utils.UploadPodName(pvc), f.Namespace.Name, time.Second*20)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(deleted).To(BeTrue())
+	})
+
+	It("Should create upload pod in namespace with quota", func() {
+		f.CreateQuotaInNs(int64(1), int64(1024*1024*1024), int64(2), int64(2*1024*1024*1024))
+		By("Creating PVC with upload target annotation")
+		pvc, err = f.CreatePVCFromDefinition(utils.UploadPVCDefinition())
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Set up port forwarding")
+		uploadProxyURL, portForwardCmd, err = startUploadProxyPortForward(f)
+		Expect(err).ToNot(HaveOccurred())
+		By("Verify PVC annotation says ready")
+		found, err := utils.WaitPVCPodStatusReady(f.K8sClient, pvc)
```

Comment:
The quota only effects whether the pod gets created, right?  So why do the rest of this?

Reply:
True, I just wanted it to do a complete operation while inside the quota, but you are right technically we just need to verify the pod is created.